source/_components/alarmdecoder.markdown

```diff
@@ -44,16 +44,50 @@ alarmdecoder:
-- **panel_display** (*Optional*): Create a sensor called sensor.alarm_display to match the Alarm Keypad display. Default: `off`
-- **zones** (*Optional*): AlarmDecoder has no way to tell us which zones are actually in use, so each zone must be configured in Home Assistant. For each zone, at least a name must be given. For more information on the available zone types, take a look at the [Binary Sensor](/components/binary_sensor.alarmdecoder/) docs. *Note: If no zones are specified, Home Assistant will not load any binary_sensor components.*
-- **rfid** (*Optional*): The RF serial-number associated with RF zones. Providing this field allows Home Assistant to associate raw sensor data to a given zone, allowing direct monitoring of the state, battery, and supervision status.
-- **relayaddr** (*Optional*): Address of the relay expander board to associate with the zone. (ex: 12, 13, 14, or 15). Typically used in cases where a panel will not send bypassed zones such as motion during an armed home state, the Vista 20P is an example of this. Alarmdecoder can emulate a zone expander board and the panel can be programmed to push zone events to this virtual expander. This allows the bypassed zone binary sensors to be utilized. One example is using bypassed motion sensors at night for motion-based automated lights while the system is armed with the motion sensor bypassed.
-- **relaychan** (*Optional*): Channel of the relay expander board to associate with the zone. (ex: 1, 2, 3, or 4)
+{% configuration %}
+type:
+  description: "The type of AlarmDecoder device, optional values: `socket`, `serial` or `USB`"
+  required: true
+  type: map
+host:
+  description: The IP address of the AlarmDecoder device on your home network, if using socket type. 
+  required: false
+  default: "`localhost`"
+  type: map
+port:
+  description: The port of the AlarmDecoder device on your home network, if using socket type.
+  required: false
+  default: "`10000`"
+  type: integer
+path:
+  description: The path of the AlarmDecoder device, if using socket type.
+  required: false
+  default: "`/dev/ttyUSB0`"
+  type: string
+baudrate:
+  description: The baud rate of the AlarmDecoder device, if using serial type.
+  required: false
+  default: "`115200`"
+  type: string
+panel_display:
+  description: Create a sensor called sensor.alarm_display to match the Alarm Keypad display.
+  required: false
+  default: "`off`"
+  type: boolean
+zones:
+  description: "AlarmDecoder has no way to tell us which zones are actually in use, so each zone must be configured in Home Assistant. For each zone, at least a name must be given. For more information on the available zone types, take a look at the [Binary Sensor](/components/binary_sensor.alarmdecoder/) docs. *Note: If no zones are specified, Home Assistant will not load any binary_sensor components.*"
+  required: false
+  type: string
+rfid:
```

Comment:
This is actually a key for one of the `zones`. The rest too.

Reply:
Getting a bit too much work for this PR 

For type / zone: I am not sure what map / key mean or how to use them

for "baudrate" although the value appears to be integer the code states:
`    vol.Optional(CONF_DEVICE_BAUD, default=DEFAULT_DEVICE_BAUD): cv.string})`

* I see why people chose to move past this particular component and work on others 🙄

Reply:
Me neither. I find the standard a bit fuzzy about that – home-assistant/developers.home-assistant#107